### PR TITLE
Fixes #10823: Select system techniques and generate correct policies based on agent type

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/10823

This add the notion of AgentType 

- in the technique metadata.xml (parse <AGENT type="..."> and add structure AgentConfig which holds Bundle, Files and Templates (with a default behavior to handle the case were <AGENT> is not present, default-ing to cfengine-community + cfengine-nova)
- in the generation, and especially in the Bundle generation. 

We also add then new AgentSpecificLogic.scala which holds how bundle and files for agent, and is ready to be extensible and plugable. 
There is also some renaming (node -> xml in parsing for ex, to keep reader sanity, or  	Cf3PromisesFileWriterService -> PolicyWriterService)

Must be merged before #1670 and #1669

### CODE REVIEW

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/test/resources/configuration-repository/techniques/dsc-agent/dsc-agent/1.0/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<TECHNIQUE name="DSC Agant">
```
Comment (ncharles) [inline rudder-core/src/test/resources/configuration-repository/techniques/dsc-agent/dsc-agent/1.0/metadata.xml]: DSC Agent

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala [hunk 2]
```diff
@@ -144,14 +143,14 @@ object NodeConfigurationHash {
      */
     val nodeInfoHashValue = {
       val i = nodeConfig.nodeInfo
-      List(
+      List[Int](
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala]: AFAI recall, I had a warning by the compiler if not doing so

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala (hunk 2 above)
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala]: ok, thanks

File: rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala [hunk 3]
```diff
@@ -231,7 +241,7 @@ class WriteSystemTechniquesTest extends Specification with Loggable with BoxSpec
 
   //////////// set-up auto test cleaning ////////////
   override def afterAll(): Unit = {
-    if(System.getProperty("tests.clean.tmp") != "false") {
+    if(false) { // System.getProperty("tests.clean.tmp") != "false") {
```
Comment (ncharles) [inline rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala]: i think you forgot this one

**Comment by fanf** [comment]
PR rebased

**Pull request merged**